src/search.cpp

```diff
@@ -606,22 +605,23 @@ Value Search::Worker::search(
     // Step 4. Transposition table lookup.
     excludedMove = ss->excludedMove;
     posKey       = pos.key();
-    tte          = tt.probe(posKey, ss->ttHit);
-    ttValue   = ss->ttHit ? value_from_tt(tte->value(), ss->ply, pos.rule50_count()) : VALUE_NONE;
-    ttMove    = rootNode  ? thisThread->rootMoves[thisThread->pvIdx].pv[0]
-              : ss->ttHit ? tte->move()
-                          : Move::none();
-    ttCapture = ttMove && pos.capture_stage(ttMove);
+
+    TTData tte = tt.probe(posKey, ss->ttHit);
+    ttValue    = ss->ttHit ? value_from_tt(tte.value(), ss->ply, pos.rule50_count()) : VALUE_NONE;
```

Comment:
I can try this but then it become similar to Sopels test but for me this approach seems clearer. The additional values are not used in all places like in qsearch and bundles for my feeling to much search related logic there. For me its a simply immutable wrapper for the tt entry which should not have more logic. But i will make a version and test so maintainers have more different versions for decision. The question if this version has then the same problem as Sopel's one to pass.

Reply:
I think we are talking about different things.. I was just asking to about moving
`Move ttMove` from here https://github.com/official-stockfish/Stockfish/blob/e23913a9a5489f766558665a8951fad221e23761/src/search.cpp#L550
to here https://github.com/official-stockfish/Stockfish/blob/e23913a9a5489f766558665a8951fad221e23761/src/search.cpp#L611

Reply:
or perhaps we give `ttMove`, `ttValue` and `ttCapture` different names? Yes, these variables are used in search only while constructed based on information of a TT entry. With this approach I'd like to leave these outside `TTData`, but if one can wrap all variables into a single struct without losing readability/performance, that's still worth to consider.

Reply:
Mh different names, dunno which ones lol? 
I was just saying all this since we now initialize the tt data directly we might as well just move the other tt related logic down to where it also first assigned. Basically moving away from this legacy C requirement where variables had to be declared at the start of a function.

Reply:
@Disservin 
That i misunderstand. Yes that can we do. In my C++ code i prefer this too(if possible because of scope). 'Init at declaration'. Its anyway safer by avoiding addicently undefined behavior.

Reply:
Refactored PR with @Disservin  and @MinetaS  comments.